app/assets/javascripts/modules/organisation-list-filter.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-var */
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function FilterOrganisations ($module) {
+    this.$module = $module
+    this.filterTimeout = null
+    this.form = this.$module.querySelector('[data-filter="form"]')
+    this.orgResults = this.$module.querySelector('#organisations_search_results')
+  }
+
+  FilterOrganisations.prototype.init = function () {
+    this.$module.filterList = this.filterList.bind(this)
+    // Form should only appear if the JS is working
+    this.form.classList.add('filter-organisations-list__form--active')
+    this.results = document.createElement('div')
+    this.results.classList.add('filter-organisations-list__results', 'govuk-heading-m', 'js-search-results')
+    this.results.setAttribute('aria-live', 'polite')
+    this.results.innerHTML = this.countInitialDepartments() + ' results found'
+    this.orgResults.insertBefore(this.results, this.orgResults.firstChild)
+
+    // We don't want the form to submit/refresh the page on enter key
+    this.form.onsubmit = function () { return false }
+
+    this.form.addEventListener('keyup', function (e) {
+      var searchTerm = e.target.value
+      clearTimeout(this.filterTimeout)
+      this.filterTimeout = setTimeout(function () {
+        this.$module.filterList(searchTerm)
+      }.bind(this), 200)
+    }.bind(this))
+  }
+
+  FilterOrganisations.prototype.filterList = function (searchTerm) {
+    var itemsToFilter = this.$module.querySelectorAll('[data-filter="item"]')
+    var listsToFilter = this.$module.querySelectorAll('[data-filter="list"]')
+    for (var i = itemsToFilter.length - 1; i >= 0; i--) {
```

Comment:
How come we're counting down here? The typical convention I've observed in our JS is to count up when iterating. I can't see anything in the loop itself that would benefit from reverse the count. Is it a bother to swap this around to keep with convention?

```suggestion
    for (var i = 0; i <= itemsToFilter.length; i++) {
```

Reply:
You're right this is odd. We originally did a reverse loop because we thought we were going to be removing things from the array, but in the end all we needed was to add a class.

I'll change the loop and fix up.